domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/metadata/ClaimantResponseMetadata.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.cmc.domain.models.metadata;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ClaimantResponse;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ClaimantResponseType;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.CourtDetermination;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.DecisionType;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.FormaliseOption;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ResponseAcceptation;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ResponseRejection;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import static java.util.function.Predicate.isEqual;
+
+@Getter
+@EqualsAndHashCode
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+class ClaimantResponseMetadata {
+    private final LocalDateTime claimantRespondedAt;
+    private final ClaimantResponseType claimantResponseType;
+    private final Boolean fullPaymentReceived;
+    private final Boolean settleForAmount;
+    private final PaymentPlanMetadata claimantPaymentPlan;
+    private final PaymentPlanMetadata courtPaymentPlan;
+    private final DecisionType courtDecision;
+    private final FormaliseOption formaliseOption;
+    private final Boolean mediation;
+
+    static ClaimantResponseMetadata fromClaim(Claim claim) {
+        final Optional<ClaimantResponse> optionalClaimantResponse = claim.getClaimantResponse();
+        if (!optionalClaimantResponse.isPresent()) {
+            return null;
+        }
+
+        final ClaimantResponse claimantResponse = optionalClaimantResponse.get();
+        final LocalDateTime claimantRespondedAt = claim.getClaimantRespondedAt().orElse(null);
+        final Optional<YesNoOption> optionalSettleForAmount = claimantResponse.getSettleForAmount();
+        final Boolean settleForAmount = optionalSettleForAmount.isPresent()
+            ? optionalSettleForAmount.filter(isEqual(YesNoOption.YES)).isPresent()
+            : null;
+        final Optional<BigDecimal> optionalAmountPaid = claimantResponse.getAmountPaid();
+        final Boolean fullPaymentReceived = optionalAmountPaid.isPresent()
+            ? optionalAmountPaid.filter(isEqual(claim.getTotalAmountTillToday())).isPresent()
+            : null;
+        final ClaimantResponseType claimantResponseType = claimantResponse.getType();
+
+        if (claimantResponse instanceof ResponseAcceptation) {
```

Comment:
Can we please use `claimantResponse.getType()` in switch instead of `instanceof` with `if`?
We can also extract out the individual methods for each type to reduce the complexity of this method.

Reply:
Thanks 👍